### ISSUE TITLE
connectors: ALTER SECRET causes a panic if it has been used in a connector

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#12467]
### What version of Materialize are you using?

v0.26.1-dev (d12bfe583)

### How did you install Materialize?

Docker image

### What is the issue?

If a secret is used in a connector, renaming it causes a panic . To reproduce:

1. Remove `test/kafka-ssl/with-options-errors.td` and start the `kafka-ssl` test
2. Run
```
materialize=>  alter secret csr_pass rename to csr_pass2;
server closed the connection unexpectedly
```

We allow tables to be renamed and we will rewrite the definitions of any views that reference that table.

### Relevant log output

```source
2022-05-16T17:14:15.087327Z ERROR panic: internal error: entered unreachable code: Internal error: only catalog items need to update item refs
thread: coordinator
location: src/sql/src/ast/transform.rs:99:14
version: 0.26.1-dev (d12bfe5832f7d4530fc435c48acf3f2123d2b94f)
backtrace:
   0: materialized::handle_panic
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/materialized/src/bin/materialized/main.rs:865:21
   1: core::ops::function::Fn::call
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:70:5
   2: std::panicking::rust_panic_with_hook
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:702:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:588:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
   5: rust_begin_unwind
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   6: core::panicking::panic_fmt
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   7: mz_sql::ast::transform::create_stmt_rename_refs
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/sql/src/ast/transform.rs:99:14
   8: mz_coord::catalog::CatalogItem::rename_item_refs::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/catalog.rs:1001:13
   9: mz_coord::catalog::CatalogItem::rename_item_refs
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/catalog.rs:1041:32
  10: mz_coord::catalog::Catalog<S>::transact::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/catalog.rs:2554:39
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::catalog_transact::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:4326:15
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::sequence_alter_item_rename::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:4172:58
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::sequence_plan::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:1746:62
  12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::handle_execute::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:1441:66
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::message_command::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:1043:62
  13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator<S>::serve::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:761:67
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/park/thread.rs:263:54
      tokio::coop::with_budget::{{closure}}
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:102:9
      std::thread::local::LocalKey<T>::try_with
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/local.rs:413:16
      std::thread::local::LocalKey<T>::with
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/local.rs:389:9
      tokio::coop::with_budget
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:95:5
      tokio::coop::budget
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:72:5
      tokio::park::thread::CachedParkThread::block_on
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/park/thread.rs:263:31
  15: tokio::runtime::enter::Enter::block_on
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/enter.rs:151:13
      tokio::runtime::handle::Handle::block_on
             at cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/handle.rs:355:9
  16: mz_coord::coord::serve::{{closure}}::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-03d586b0eaaf88b56-1/materialize/tests/src/coord/src/coord.rs:4783:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:122:18
  17: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/mod.rs:498:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
      std::panicking::try
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/mod.rs:497:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:227:5
  18: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
      std::sys::unix::thread::Thread::new::thread_start
             at rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys/unix/thread.rs:108:17
  19: <unknown>
  20: __clone
```

**Comment by philip-stoev** [comment]
I think Connectors need to be listed in `create_stmt_rename_refs`

**Comment by benesch** [comment]
Retroactive duplicate of #15552, which, happily, was fixed by #15563.
